server/odc-service/src/main/java/com/oceanbase/odc/service/session/initializer/ConsoleTimeoutInitializer.java

```diff
@@ -40,31 +43,55 @@ public class ConsoleTimeoutInitializer implements ConnectionInitializer {
         TIMEOUT_VARIABLES_SET.add("ob_pl_block_timeout");
     }
 
-    public ConsoleTimeoutInitializer(@NonNull Long timeouUs) {
+    public ConsoleTimeoutInitializer(@NonNull Long timeouUs, DialectType dialectType) {
         this.timeoutUs = timeouUs;
+        this.dialectType = dialectType;
     }
 
     @Override
     public void init(Connection connection) throws SQLException {
-        Set<String> variableToBeModified = new HashSet<>();
-        try (Statement stmt = connection.createStatement();
-                ResultSet resultSet = stmt.executeQuery("show session variables like '%timeout'")) {
-            while (resultSet.next()) {
-                String variableName = resultSet.getString(1);
-                if (!TIMEOUT_VARIABLES_SET.contains(variableName)) {
-                    continue;
+        if (DialectType.ORACLE.equals(this.dialectType)) {
+            try {
+                try (Statement stmt = connection.createStatement();
```

Comment:
Seems no need nested try block.

Reply:
An exception is likely to be thrown here because DBA_PROFILES requires sysdba permissions. If the user does not have this permission, an exception will be thrown.